fs.sshfs v 0.6.0

ReimarBauer opened the pull request.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/fs.sshfs```) and found it was in an excellent condition.

ReimarBauer commented:
Argh, that skip syntax I get mostly wrong

ReimarBauer commented:
found 2 valuable problems I would have excepted to get a linter hint.

ReimarBauer commented:
a later version may be compatible with windows, now we have to skip that

ReimarBauer commented:
Hi @dougalsutherland 
at that package I learned already many things. But the last problem I can't solve. I get always on any CI a traceback.

What can I do to fix this?

```
Fetching package metadata .............
Traceback (most recent call last):
  File "/Users/travis/miniconda/bin/conda-build-all", line 6, in <module>
    exit(main())
  File "/Users/travis/miniconda/lib/python3.6/site-packages/conda_build_all/cli.py", line 93, in main
    b.main()
  File "/Users/travis/miniconda/lib/python3.6/site-packages/conda_build_all/builder.py", line 258, in main
    recipe_metas = self.fetch_all_metas(build_config)
  File "/Users/travis/miniconda/lib/python3.6/site-packages/conda_build_all/builder.py", line 173, in fetch_all_metas
    recipe_metas = sort_dependency_order(recipe_metas, config=config)
  File "/Users/travis/miniconda/lib/python3.6/site-packages/conda_build_all/builder.py", line 122, in sort_dependency_order
    all_deps = [dep.split(' ', 1)[0] for dep in all_deps]
  File "/Users/travis/miniconda/lib/python3.6/site-packages/conda_build_all/builder.py", line 122, in <listcomp>
    all_deps = [dep.split(' ', 1)[0] for dep in all_deps]
AttributeError: 'dict' object has no attribute 'split'
The command "if [ -n "$GH_TOKEN" ]; then echo "Creating feedstocks from the recipe(s)."; git config --global user.name "Travis-CI on github.com/conda-forge/staged-recipes"; git config --global user.email "conda-forge@googlegroups.com"; source ./.CI/create_feedstocks; else echo "Building all recipes."; source ./.CI/build_all; fi" exited with 1.
Done. Your build exited with 1.
```

ReimarBauer commented:
May be it is something similar to https://github.com/tensorflow/models/issues/1587

ocefpaf commented on recipes/fs.sshfs/meta.yaml:
```diff
@@ -0,0 +1,125 @@
+
+requirements:
+  build:
+    - skip: true  # [win]
```
This is in the wrong place and may be the culprit of the failure you are seeing.

ReimarBauer replied:
seems I am blind. Thx, lets see.

ocefpaf commented on recipes/fs.sshfs/meta.yaml:
```diff
@@ -0,0 +1,125 @@
+  license_file: '{{ environ["RECIPE_DIR"] }}/COPYING'
+  summary: 'Pyfilesystem2 over SSH using paramiko'
+  description: >
+    Usage
```
I strongly encourage you do make this shorter. No need to replicate docs in the description.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/fs.sshfs```) and found some lint.

Here's what I've got...


For **recipes/fs.sshfs**:

 * Failed to even lint the recipe (might be a conda-smithy bug) :cry:

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/fs.sshfs```) and found it was in an excellent condition.

jakirkham commented on recipes/fs.sshfs/meta.yaml:
```diff
@@ -0,0 +1,47 @@
+
+build:
+  skip: true  # [win]
+  noarch: python
```
Can't be `noarch` if it needs selectors unfortunately. So this will need to be dropped.

ReimarBauer commented:
@jakirkham @ocefpaf anything else ;) ?

dougalsutherland commented on recipes/fs.sshfs/meta.yaml:
```diff
@@ -0,0 +1,46 @@
+    - fs
+    - six
+    - paramiko
+    - configparser  # [py27 and py34]
```
This will never trigger; you probably want `[py27 or py34]`.

ReimarBauer commented:
Hi @dougalsutherland 
can I get merged this?

dougalsutherland commented:
Sorry for not merging earlier. I didn't because I noticed that the pypi package pins `fs` and some other dependencies, but this recipe doesn't. I'll make an issue in the feedstock.
